### ISSUE TITLE
NullCheckFilter: no rows processed in large oracle database

### ISSUE THREAD

**Issue by jhorcicka** [datacleaner/DataCleaner#1424]
In a simple job consisting of "oracle datastore", "null check filter" and "csv writer (or value distribution)" where the table contains "many" rows (cca 200000), the results are only produced when both branches of NullCheckFilter are used. 

When I use only "NOT_NULL" then there is no result. 

I tried that on the same remote oracle database but on a different table with only a few records (cca 50) and it worked fine. I also tried that on my/local postgres datastore with 300000 rows and it worked fine as well. So, it seems the problem is related to oracle database table with many records. 

![null-filter-issue-01](https://cloud.githubusercontent.com/assets/13213915/16621066/3fd30590-4395-11e6-99f2-582f5efafbd5.png)

![null-filter-issue-02](https://cloud.githubusercontent.com/assets/13213915/16621065/3fcf261e-4395-11e6-8027-36aaf9e66706.png)

![null-filter-issue-04](https://cloud.githubusercontent.com/assets/13213915/16621067/3fd3612a-4395-11e6-8d01-38e80f56bdbe.png)

**Comment by LosD** [comment]
My guess would be that this is either some wild race condition, or a MetaModel query optimization issue, but we probably need more research first.

@kaspersorensen have you ever seen something as fantastically weird as this?

**Comment by LosD** [comment]
With the blackouts, I guess there's confidential information in there, but if you can somehow scrub the execution log for that information, it would help at getting more eyes on it (preferably with column names changed to i.e. col_1, col_2, col_3 instead of all being named xxxxx,xxxxx,xxxxx so we can cross check column names in different rows)

**Comment by kaspersorensen** [comment]
I haven't seen it either no. And I agree that most likely this is related to the query optimizable nature of null check filter.

If you can reproduce then try to turn on Debug logging for org.datacleaner.job.runner to see what query gets fired in the end.

**Comment by jhorcicka** [comment]
![execution_log](https://cloud.githubusercontent.com/assets/13213915/16622282/e3221e76-4399-11e6-9191-bf168e33668f.png)
![filter_config](https://cloud.githubusercontent.com/assets/13213915/16622284/e3258f3e-4399-11e6-8d6f-908f9befc4cd.png)
![value_distribution_config](https://cloud.githubusercontent.com/assets/13213915/16622283/e3254f74-4399-11e6-88a1-c90c6d6408af.png)

**Comment by LosD** [comment]
Yeah okay, not much gained there. Probably better use @kaspersorensen's suggestion.

**Comment by jhorcicka** [comment]
I am not going to post the whole debug output but the problem is indeed in the final SQL query which ends with:
```
... FROM <myTableName> WHERE <dateColumn> IS NOT NULL AND (<dateColumn> <> '' OR <dateColumn> IS NULL)
```
